src/octoprint/server/api/timelapse.py

```diff
@@ -113,6 +116,21 @@ def setTimelapseConfig():
 				else:
 					return make_response("Invalid value for interval: %d" % interval)
 
+		if "retractionZHop" in request.values:
+                        config["options"] = {
+                                "retractionZHop": 0
+                        }
+
+                        try:
+                                retractionZHop = float(request.values["retractionZHop"])
+                        except ValueError:
+                                return make_response("Invalid value for retraction Z-Hop: %r" % request.values["retractionZHop"])
+                        else:
+                                if retractionZHop > 0:
```

Comment:
If I follow correctly, in the meantime for those of us on master, if simply switch to "On Z-Change", set a retraction z-hop to something other than zero, save settings, change it back to zero, and save again it should work fine.